src/app/submit.service.ts

```diff
@@ -1,20 +1,31 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable, of } from 'rxjs';
-import { mergeMap } from 'rxjs/operators';
+import { mergeMap, catchError } from 'rxjs/operators';
 import { OpenBabelData } from './OpenBabelData';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SubmitService {
-  readonly submitUrl: string = "/submit/";
-  data: OpenBabelData = {inputString: "", inputFormat: "", outputFormat: "",
+  private readonly submitUrl: string = "/submit/";
+  private readonly blankData: OpenBabelData = {inputString: "", inputFormat: "", outputFormat: "",
     additionalOptions: ""};
+  data: OpenBabelData = this.blankData;
```

Comment:
When do the fields ever get updated from the user input?  Seems like the the data will always be blank when you make the patch call.

Reply:
ngModel in the components. This is the same code I showed yesterday that sent the correct data to flask.